tests/ci/run_minimal_tests.sh

```diff
@@ -23,3 +23,20 @@ build_and_run_minimal_test -DDISABLE_GO=ON -DDISABLE_PERL=ON -DCMAKE_BUILD_TYPE=
 echo "Testing shared lib AWS-LC without Perl/Go in release mode."
 build_and_run_minimal_test -DDISABLE_GO=ON -DDISABLE_PERL=ON -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release
 
+# Special build options
+
+## build option: MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX
+echo "Testing static lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in release mode."
+build_and_run_minimal_test -DCMAKE_BUILD_TYPE=Release -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing shared lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in release mode."
+build_and_run_minimal_test -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing static lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in debug mode."
+build_and_run_minimal_test -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing shared lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in debug mode."
+build_and_run_minimal_test -DBUILD_SHARED_LIBS=1 -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing shared lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in [FIPS, release] mode."
+build_and_run_minimal_test -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON -DFIPS=1
```

Comment:
This dimension seems a combination of different build flags. What it's testing?

If FIPS is needed, replace `build_and_run_minimal_test` with `fips_build_and_test` and move to `tests/ci/run_fips_tests.sh`. But this may hit some issues because `fips_build_and_test` runs additional tests that `build_and_run_minimal_test` does not cover.